lib/bambora/json_request.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+module Bambora
+  class JSONRequest
+    attr_reader :client
+
+    def initialize(client)
+      @client = client
+      @path = ''
+    end
+
+    def request(method:, path:, params: {}, body: {})
+      resp = client.request(method: method, path: path, params: params, body: body.to_json.to_s)
+      return parse_response(resp) if resp.status <= 300
```

Comment:
i see here a problem - what if it is 404. As a json client i want to get a response with 404 and empty collection.
instead i get a generic error which i have to parse(raw text) to interpret it something meaningful in my app.
https://jsonplaceholder.typicode.com/posts/1 - this returns me an object and this returns me an empty object
https://jsonplaceholder.typicode.com/posts/12123123123 with a status in the response header.

Reply:
@bilogub Do you think we should parse all 4xx level errors and only raise an erro on 5xx and 3xx?

Reply:
@cassidycodes i think i gave a falsey option in my prev comment regading possible solution here and misled you, sorry.
In nutshell we should relay anything we get from api back to our client app(as long as we don't have our own API doc to rely on - we are only sitting in the middle between API and client app passing bytes there and back) - that said parse response in begin/rescue block and pass it up the chain. In case we end up in rescue block compose 50x server response in json format. What do you think?

Reply:
Sounds good! I just pushed a change that passes a hash with the error back to the user.